### PR TITLE
Fix build warnings

### LINKED ISSUE #65: Remove documentation build warnings
Since [updating the configuration file](https://github.com/Concordium/concordium-net-sdk/pull/64) to include documentation, several warnings for missing documentation and formatting are now shown during compilation.

We need to fix all the warnings and ideally have the CI build [treat warnings as errors](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/errors-warnings#treatwarningsaserrors) for future PRs.

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Closes #65

Building the project produces a lot of warnings, most are about missing documentation and end up hiding the more important ones.
This PR fixes every warning, but does not include full support for Concordium node v7 API, this was [added here instead](https://github.com/Concordium/concordium-net-sdk/pull/102)

### CODE REVIEW

File: src/Types/BakerEvent.cs [hunk 1]
```diff
@@ -51,6 +51,7 @@ internal static IBakerEvent From(BakerEvent bakerEvent) =>
                 ),
             BakerEvent.EventOneofCase.None =>
                 throw new MissingEnumException<BakerEvent.EventOneofCase>(bakerEvent.EventCase),
+            BakerEvent.EventOneofCase.DelegationRemoved => throw new NotImplementedException(),
```
Comment (Victor-N-Suadicani) [inline src/Types/BakerEvent.cs:54]: Why? A comment to explain why this isn't implemented would be nice.

File: src/Types/BlockHashInput.cs [hunk 2]
```diff
@@ -7,6 +7,9 @@ namespace Concordium.Sdk.Types;
 /// </summary>
 public interface IBlockHashInput
 {
+    /// <summary>
+    /// Convert this into type expected by the gRPC.
```
Comment (Victor-N-Suadicani) [inline src/Types/BlockHashInput.cs:11]: Perhaps:
```suggestion
    /// Convert this into the type expected by the gRPC API.
```

File: src/Types/BlockHeight.cs [hunk 3]
```diff
@@ -40,6 +46,9 @@ public BlocksAtHeightRequest Into() =>
 /// </param>
 public sealed record RelativeHeight(ulong Height, uint GenesisIndex, bool Restrict) : IBlockHeight
 {
+    /// <summary>
+    /// Convert an RelativeHeight into a request for blocks at height.
```
Comment (Victor-N-Suadicani) [inline src/Types/BlockHeight.cs:50]:
```suggestion
    /// Convert a RelativeHeight into a request for blocks at height.
```

Also request for blocks at what height? I guess I'm missing some context but the sentence just sounds weird. Same in the other places.

File: src/Types/CcdAmount.cs [hunk 4]
```diff
@@ -136,10 +136,27 @@ public static bool TryDeserial(ReadOnlySpan<byte> bytes, out (CcdAmount? Amount,
...
+    public static CcdAmount Max(CcdAmount first, CcdAmount second)
+    {
+        if (first < second)
+        {
+            return second;
+        }
+        else
+        {
+            return first;
+        }
```
Comment (Victor-N-Suadicani) [inline src/Types/CcdAmount.cs:151]: Does this not work? (maybe you need an explicit `return` in front, not sure)
```suggestion
        first < second ? second : first
```

File: src/Types/DelegationEvent.cs [hunk 5]
```diff
@@ -27,6 +27,7 @@ internal static IDelegationEvent From(DelegationEvent delegationEvent) =>
                 DelegationAdded.From(delegationEvent.DelegationAdded),
             DelegationEvent.EventOneofCase.DelegationRemoved =>
                 DelegationRemoved.From(delegationEvent.DelegationRemoved),
+            DelegationEvent.EventOneofCase.BakerRemoved => throw new NotImplementedException(),
```
Comment (Victor-N-Suadicani) [inline src/Types/DelegationEvent.cs:30]: Same here, why is this not implemented?

File: src/Types/UpdatePayload.cs [hunk 6]
```diff
@@ -16,28 +16,51 @@ public interface IUpdatePayload
 /// </summary>
 public enum UpdateType
 {
+    /// <summary> Update of protocol version. </summary>
```
Comment (Victor-N-Suadicani) [inline src/Types/UpdatePayload.cs:19]: I love doc comments more than the average person probably but a lot of these doc comments really add nothing. Is there nothing more to add? Or could we not disable whatever requirement is forcing these i.e. akin to Rust's `#[allow(missing_docs)]`?

**Review by Victor-N-Suadicani — APPROVED** [review]
Look alright, just a few comments but nothing blocking.

File: src/Types/BakerEvent.cs (hunk 1 above)
Comment (limemloh) [reply src/Types/BakerEvent.cs:54]: I tried to split up changes related to fixing build warnings and changes due to P7.
It is implemented in https://github.com/Concordium/concordium-net-sdk/pull/102

File: src/Types/BlockHeight.cs (hunk 3 above)
Comment (limemloh) [reply src/Types/BlockHeight.cs:50]: I change the doc comment to make it more clear

File: src/Types/UpdatePayload.cs (hunk 6 above)
Comment (limemloh) [reply src/Types/UpdatePayload.cs:19]: I don't think there is a lot more to add to this.
To me, this still seems better than disabling the warning.

File: src/Types/DelegationEvent.cs (hunk 5 above)
Comment (limemloh) [reply src/Types/DelegationEvent.cs:30]: It is implemented in https://github.com/Concordium/concordium-net-sdk/pull/102

**Pull request merged**